diff --git lib/prism/serialize.rb lib/prism/serialize.rb
index ae94d46a..24ff763c 100644
--- a/lib/prism/serialize.rb
+++ b/lib/prism/serialize.rb
@@ -8,7 +8,7 @@
 require "stringio"

 # Polyfill for String#unpack1 with the offset parameter.
-if String.instance_method(:unpack1).parameters.none? { |_, name| name == :offset }
+if String.instance_method(:unpack1).arity == 1
   String.prepend(
     Module.new {
       def unpack1(format, offset: 0) # :nodoc:
